### PR TITLE
Add link to Darwin patient view (issue #1111)

### PR DESCRIPTION
**Pull request by angelicaochoa**
Darwin link only appears for authorized users to patient's clinical data.

@n1zea144  Current link is indicated by bold green text - is there a logo/format that should be used instead?

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/portal/servlet/CheckDarwinAccess.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+/**
+ *
+ * @author ochoaa
+ */
+public class CheckDarwinAccess {
+    private static String DARWIN_AUTH_URL;
+    private static String DARWIN_RESPONSE_URL;
+    
+    public static String checkAccess(String userName, String patientId){
```
Comment (n1zea144) [inline core/src/main/java/org/mskcc/cbio/portal/servlet/CheckDarwinAccess.java]: We also want to take into account the cancer study - linking to darwin should only be relevant to mskimpact study/patients (we actually have many merged studies which contain mskimpact, but no easy way to determine that).  Also, we don't want to provide a link to darwin if the user came from the CIS system (see me to explain this).

File: core/src/main/java/org/mskcc/cbio/portal/servlet/CheckDarwinAccess.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
+/*
```
Comment (n1zea144) [inline core/src/main/java/org/mskcc/cbio/portal/servlet/CheckDarwinAccess.java]: We should move this class to the Util package.

File: core/src/main/java/org/mskcc/cbio/portal/model/DarwinAccess.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
+/*
```
Comment (n1zea144) [inline core/src/main/java/org/mskcc/cbio/portal/model/DarwinAccess.java]: We should make this class a nested class within the CheckDarwinAccess class.

**Comment by angelicaochoa** [comment]
@n1zea144 done and done

**Comment by morungos** [comment]
Wonderful that you're doing this, although I'd really like it to be through a plugin some day. Can we maybe use this as a test case for that?

**Comment by n1zea144** [comment]
@morungos Yes, me too.  In the meantime, if you have any suggestions for insulating this from the general codebase, let us know.

**Comment by angelicaochoa** [comment]
@n1zea144 @jjgao PR should be ready although right now the link to the darwin patient view is indicated by bold green text. Wasn't sure what should be used instead.

**Comment by morungos** [comment]
@n1zea144 I'm now thinking war overlays might do most, if not all of this. http://maven.apache.org/plugins/maven-war-plugin/overlays.html. Although it'll be easier once we've slimmed down a bunch of the configuration. Certainly this will allow the Java and web components to be overridden by an individual deployment without pushing them into core.

**Comment by n1zea144** [comment]
@morungos I'm catching up on your comments about war overlays in issues #1299.  We are certainly open to refactoring this code using overlays to prevent custom code from entering into the core.

**Pull request merged**